### PR TITLE
Fix broken links

### LINKED ISSUE #5986: 404 at /datacenter/dtr/2.4/configure/garbage-collection/
On the page Docker Trusted Registrty overview under [Efficieny](https://docs.docker.com/datacenter/dtr/2.4/guides/#efficiency) the link to [clean up unreferenced manifests and layers.](https://docs.docker.com/datacenter/dtr/2.4/configure/garbage-collection/) is broken and gives 404.

### PR DESCRIPTION
**Pull request by joaofnfernandes**
Fixes #5986 and a bunch of other links that were broken.

### CODE REVIEW

File: datacenter/dtr/2.4/guides/index.md [hunk 1]
```diff
@@ -34,10 +34,10 @@ and metadata such that if a machine fails, DTR continues to operate and can be r
 
 ## Efficiency
 
-DTR has the ability to [cache images closer to users](../admin/configure/deploy-caches/index.md)
+DTR has the ability to [cache images closer to users](admin/configure/deploy-caches/index.md)
```
Comment (gbarr01) [inline datacenter/dtr/2.4/guides/index.md]: @joaofnfernandes , is this link going  to work? It doesn't for me.

File: datacenter/dtr/2.4/guides/index.md [hunk 2]
```diff
@@ -52,18 +52,18 @@ DTR has a built in security scanner that can be used to discover what versions
...
+[unprecedented insight into your exposure to known security threats](user/manage-images/scan-images-for-vulnerabilities.md).
 
 ## Image signing
 
 DTR ships with [Notary](/notary/getting_started.md)
 built in so that you can use
 [Docker Content Trust](/engine/security/trust/content_trust.md) to sign
 and verify images. For more information about managing Notary data in DTR see
-the [DTR-specific notary documentation](user/manage-images/sign-images/manage-trusted-repositories.md).
+the [DTR-specific notary documentation](user/manage-images/sign-images/sign-images/index.md).
```
Comment (gbarr01) [inline datacenter/dtr/2.4/guides/index.md]: I think this should be: `user/manage-images/sign-images/`

File: datacenter/dtr/2.4/guides/index.md (hunk 1 above)
Comment (gbarr01) [reply datacenter/dtr/2.4/guides/index.md]: This works: `admin/configure/deploy-caches/`

Comment (joaofnfernandes) [reply datacenter/dtr/2.4/guides/index.md]: Both urls should be valid. I've tested the one ending in `.md` with netlify and and local build and it seems to work. Maybe I'm missing something 🤔

Comment (gbarr01) [reply datacenter/dtr/2.4/guides/index.md]: Probably not. Go for it.

**Pull request merged**